### PR TITLE
Remove pricing page deprecated variant SPP

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR removes the code related to the SPP variant of the pricing page, which didn't lead to a conclusive experiment.

### Testing instructions

- Download the PR and run cloud or Calypso
- Visit the pricing page locally
- Open a new tab and visit the pricing page in staging or production
- Check that both pages look and behave similarly
- Verify there's no error in the console

### CODE REVIEW

File: client/lib/plans/features-list.js [hunk 1]
```diff
@@ -1068,11 +1065,7 @@ export const FEATURES_LIST = {
 
 	[ constants.FEATURE_BACKUP_REALTIME_V2 ]: {
 		getSlug: () => constants.FEATURE_BACKUP_REALTIME_V2,
-		getTitle: ( variation ) =>
-			( {
-				i5: i18n.translate( 'Backup (real-time, off-site)' ),
-				spp: i18n.translate( 'Backup (real-time, off-site)' ),
-			}[ variation ] || i18n.translate( 'Automated real-time site backups' ) ),
+		getTitle: () => i18n.translate( 'Backup (real-time, off-site)' ),
```
Comment (monsieur-z) [inline client/lib/plans/features-list.js]: No other version that i5 at the moment.

File: client/lib/plans/plans-list.js [hunk 2]
```diff
@@ -391,14 +391,7 @@ const getPlanBusinessDetails = () => ( {
 const getPlanJetpackSecurityDailyDetails = () => ( {
 	group: constants.GROUP_JETPACK,
 	type: constants.TYPE_SECURITY_DAILY,
-	getTitle: ( variation ) =>
-		( {
-			spp: translate( 'Jetpack Security' ),
-		}[ variation ] || translate( 'Security {{em}}Daily{{/em}}', { components: { em: <em /> } } ) ),
-	getButtonLabel: ( variation ) =>
```
Comment (monsieur-z) [inline client/lib/plans/plans-list.js:398]: `getButtonLabel` is optional and can safely be removed.

**Comment by mattgawarecki** [comment]
I took the liberty of resolving the conflicts from my recent experiment being deployed. :-)

**Review by mattgawarecki — APPROVED** [review]
Code changes look good, and everything tested well for me. Nice cleanup! It'll be great to pair this with #50100 😁

**Pull request merged**